yb-voyager/cmd/importData.go

```diff
@@ -915,54 +896,112 @@ func doOneImport(task *SplitFileImportTask, connPool *tgtdb.ConnectionPool) {
 	}
 }
 
-func executeSqlFile(file string) {
-	log.Infof("Execute SQL file %q on target %q", file, target.Host)
-	connectionURI := target.GetConnectionUri()
-	conn, err := pgx.Connect(context.Background(), connectionURI)
+func newTargetConn() *pgx.Conn {
+	conn, err := pgx.Connect(context.Background(), target.GetConnectionUri())
 	if err != nil {
 		utils.WaitChannel <- 1
 		<-utils.WaitChannel
 		utils.ErrExit("connect to target db: %s", err)
 	}
-	defer conn.Close(context.Background())
 
+	setTargetSchema(conn)
+	return conn
+}
+
+func setTargetSchema(conn *pgx.Conn) {
 	if sourceDBType != POSTGRESQL && target.Schema != YUGABYTEDB_DEFAULT_SCHEMA {
```

Comment:
Negating this condition will make it more readable (remember the `De Morgan's Theorem`?? 🙂 ):
```
  if sourceDBType == POSTGRESQL || target.Schema == YUGABYTEDB_DEFAULT_SCHEMA {
    // For PG, schema name is already included in the object name.
    // No need to set schema if importing in the default schema.
    return
  }
  // Proceed to running SET SCHEMA.
```

Reply:
Yeah nice reference.